client/file-browser/file-browser-entries.tsx

```diff
@@ -94,30 +100,117 @@ export function FolderEntry({
   )
 }
 
-const FileEntryContainer = styled(EntryContainer)`
+const SelectButton = styled(TextButton)<{ $focused: boolean }>`
+  margin: 0 8px;
+
+  ${props =>
+    !props.$focused
+      ? css`
+          display: none;
+        `
+      : ''}
+`
+
+const StyledTooltip = styled(Tooltip)<{ $focused: boolean }>`
+  ${props =>
+    !props.$focused
+      ? css`
+          display: none;
+        `
+      : ''}
+`
+
+const FileEntryContainer = styled(EntryContainer)<{ $clickable: boolean }>`
+  ${props =>
+    !props.$clickable
+      ? css`
+          cursor: auto !important;
+        `
+      : ''}
+
   & ${EntryIcon} {
     background: ${blue700};
     color: ${colorTextPrimary};
   }
+
+  &:hover ${SelectButton} {
+    display: inline-table;
+  }
+  &:hover ${StyledTooltip} {
+    display: flex;
+  }
 `
 
-export function FileEntry({
-  file,
-  icon,
-  isFocused,
-  onClick,
-}: FileBrowserEntryProps & {
-  file: FileBrowserFileEntry
-  icon: React.ReactElement
-  onClick: (entry: FileBrowserFileEntry) => void
-}) {
-  return (
-    <FileEntryContainer $focused={isFocused} onClick={() => onClick(file)}>
-      <EntryIcon>{icon}</EntryIcon>
-      <InfoContainer>
-        <Subtitle1>{file.name}</Subtitle1>
-        <Caption>{longTimestamp.format(file.date)}</Caption>
-      </InfoContainer>
-    </FileEntryContainer>
-  )
-}
+const ExpandText = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`
+
+// NOTE(2Pac): This component is currently pushing the boundaries of what a generic component should
```

Comment:
The reason why I left the comment now instead of doing the restructuring is because I think I already restructured the low-hanging bad fruits in the file-browser when moving to functional components. Right now it doesn't even seem _that_ bad, and I'm decently certain that the current implementation covers most of our use cases for now, so we should probably wait before we get a new use case before doing any major restructuring (or until we decide to move the file-browser outside the activity overlays).